lib_src/src/asrc_task/asrc_task.c

```diff
@@ -376,6 +376,7 @@ DEFINE_INTERRUPT_PERMITTED(ASRC_ISR_GRP, void, asrc_processor,
         while(1){
             // Wait for block of samples. We will get the buffer index of the newly written samples from receive_asrc_input_samples_cb
             unsigned input_write_idx = (unsigned)chanend_in_byte(c_buff_idx);
+            int32_t save_timestamp = asrc_io->input_timestamp;
```

Comment:
HI, as discussed I think this should be double buffered in the asrc_io struct ideally. Although arguably this is all hidden from the user.
You did mention something about an issue with that approach?

Reply:
hey @ed-xmos, no issue as such. I'd mentioned that doing it the double buffer way would require more lines of code change since we're changing the asrc_in_out_t struct. It'll require change in lib_src as well as any applications that use it (an02003 for example). I do agree that that's a better fix so will do it the double buffer way and make the change everywhere.